modules/test/conn/python/src/connection_module.py

```diff
@@ -74,6 +74,60 @@ def __init__(self, module):
     # response = self.dhcp1_client.set_dhcp_range('10.10.10.20','10.10.10.30')
     # print("Set Range: " + str(response))
 
+  def _connection_switch_arp_inspection(self):
+    LOGGER.info('Running connection.switch.arp_inspection')
+
+    no_arp = True
+
+    # Read all the pcap files
+    packets = rdpcap(STARTUP_CAPTURE_FILE) + rdpcap(MONITOR_CAPTURE_FILE)
+    for packet in packets:
+
+      # We are not interested in packets unless they are ARP packets
+      if not ARP in packet:
+        continue
+
+      # We are only interested in packets from the device
+      if packet.src != self._device_mac:
+        continue
+
+      # Get the ARP packet
+      arp_packet = packet[ARP]
+      no_arp = False
+
+      # Check MAC address matches IP address
+      if (arp_packet.hwsrc != self._device_mac or
+          arp_packet.psrc != self._device_ipv4_addr):
+        return False, 'Device is sending false ARP response'
+
+    if no_arp:
+      return None, 'No ARP packets from the device found'
+
+    return True, 'Device uses ARP'
+
+  def _connection_switch_dhcp_snooping(self):
+    LOGGER.info('Running connection.switch.dhcp_snooping')
+
+    disallowed_dhcp_types = [2, 4, 5, 6, 9, 10, 12, 13, 15, 17]

```

Comment:
This doesn't appear like you are allowing any DHCP traffic from the device which does not match the test requirements.  Not sure what the intention here is for filtering this.

Reply:
DHCP snooping is switch functionality to prevent a device from sending DHCP server messages on the network. We cannot add a managed switch with DHCP enabled so instead this test checks that the device does not send those DHCP server messages - pretending to be a switch. The array contains all IDs of the DHCP server messages.